packages/asset-swapper/src/network/samplers/utils/pools_cache.ts

```diff
@@ -27,8 +32,17 @@ export abstract class PoolsCache {
         makerToken: string,
         timeoutMs: number = DEFAULT_TIMEOUT_MS,
     ): Promise<Pool[]> {
-        const timeout = new Promise<Pool[]>(resolve => setTimeout(resolve, timeoutMs, []));
-        return Promise.race([this._getAndSaveFreshPoolsForPairAsync(takerToken, makerToken), timeout]);
+        // Only allow one outstanding refresh per pair at a time.
```

Comment:
idk if this makes a difference but seemed reasonable and wasn't particularly hard?